src/app/core/models/server-config/server-config.mapper.ts

```diff
@@ -0,0 +1,29 @@
+import { ServerConfigData, ServerConfigDataEntry } from './server-config.interface';
+import { ServerConfig } from './server-config.model';
+
+export class ServerConfigMapper {
+  private static filterOutId(obj) {
+    return Object.keys(obj)
+      .filter(k => k !== 'id')
+      .reduce((acc, val) => ({ ...acc, [val]: obj[val] }), {});
+  }
+
+  private static mapEntries(entries: ServerConfigDataEntry[]) {
+    return entries.reduce(
+      (acc, entry) => ({
+        ...acc,
+        [entry.id]: Array.isArray(entry.elements)
+          ? ServerConfigMapper.mapEntries(entry.elements)
```

Comment:
this recursion is what for? i don't expect the configuration has more than 2 levels ...?

Reply:
Yes, currently we might only get a depth of 2 levels, but I think defining (data structure) and handling (mapping) it recursively saves quite a lot of code. The way the data coming from the server is structured, it may go deeper levels in the future.